packages/odyssey-react/src/components/Table/TableContainer.tsx

```diff
@@ -0,0 +1,50 @@
+/*!
+ * Copyright (c) 2021-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from 'react';
+import type { ReactNode, DetailedHTMLProps, HTMLAttributes } from 'react';
+
+import { tableClass } from './Table';
+
+export type Props = {
+  /**
+  * The table for this container
+  */
+  children?: ReactNode,
+  /**
+  * The visible heading for your table
+  */
+  title: ReactNode,
+} & DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>
+
+type Ref = HTMLElement;
+
+const TableContainer = React.forwardRef<Ref, Props>((props, ref) => {
+  const {
+    children,
+    className,
+    title,
+    ...rest
+  } = props;
+
+  const containerClass = `${tableClass}--figure`;
+  const combinedClassName = className ? `${className} ${containerClass}` : containerClass;
+
+  return (
+    <figure ref={ref} className={combinedClassName} {...rest}>
+      <figcaption className={`${tableClass}--figcaption`}>{title}</figcaption>
+      {children}
+    </figure>
```

Comment:
This is the expected wrapper. The `figure` allows for table to scroll horizontally instead of "squishing" on smaller screens.

Reply:
The inclusion of table prefixed classes made me think it was doing something important. Also the criteria outlined in the design docs.
"Each Table should have both a title and a caption.

The title is the visible heading for your table. A concise, descriptive title will provide users with a direct context for your data. The needed context may be missing if you rely on page headings or surrounding body copy."

Reply:
@edburyenegren-okta @andrewberg-okta I see - why not include this markup as part of `Table` then? If its providing important scroll behavior I would think we wouldn't want to expose it as an optional Container

Reply:
A few things at play for me
- Single responsibility principal - mixing the table and the container leads to complexity and confusion
- Composability - What if it there were another element accompanying the table in the container?
- Suggested as opposed to required  - The container is there for use but you may encounter a situation where it isn't necessary

Reply:
@andrewberg-okta I like these reasons! I think my question is at that point are we building `Table.Container` and `Table` as two top level components? Or is this PR about building `Table` as we originally planned? We can still use `Container` and give it a single responsibility and have it be optional by composing it internal to `Table` itself. Similar to the API @arnoldsandoval-okta [is using here for Tab](https://github.com/okta/odyssey/pull/1006/files#diff-e35526182e327bcd41e4f440481603c9582e59823d9ebfa867972ac25e32ce31R95). This may even solve for composability too assuming we can nest Containers arbitrarily.